File: python/anagram/anagram.py
def detect_anagrams(target_word, wordlist):
    anagrams = []
    for word in wordlist:
        if is_anagram(target_word, word):
            anagrams.append(word)
    return anagrams

def is_anagram(target, test):
    return test.lower() != target.lower() and sort(test) == sort(target)

def sort(word):
    return ''.join(sorted(word.lower()))

File: javascript/anagram/anagram.js
var Anagram = function(input) {
  this.input = input.toLowerCase();
  this.matches = function(possible_anagrams) {
    if (typeof arguments[0] == 'string') {
      possible_anagrams = Array.prototype.slice.call(arguments);
    }
    return possible_anagrams.filter(isAnagram);
  };
  return this;
};

function isAnagram(word) {
  if(word.toLowerCase() != input) {
    return sortWord(word) == sortWord(input);
  } else { return false; }
};

function sortWord(word) {
  return word.toLowerCase().split("").sort().join();
};

module.exports = Anagram;
